pkg/envoy/lds/inmesh.go

```diff
@@ -383,14 +383,12 @@ func (lb *listenerBuilder) getOutboundTCPFilter(upstream service.MeshService) (*
 func (lb *listenerBuilder) getOutboundFilterChainPerUpstream() []*xds_listener.FilterChain {
 	var filterChains []*xds_listener.FilterChain
 
-	upstreamServices := lb.meshCatalog.ListAllowedOutboundServicesForIdentity(lb.serviceIdentity)
-	if len(upstreamServices) == 0 {
+	dstServices := lb.meshCatalog.ListMeshServicesForIdentity(lb.serviceIdentity)
```

Comment:
This doesn't seem right, why are we altering existing behavior? If we need to build something specific for multicluster it needs to be a separate code path

Reply:
Correct, this should not program all services for the given identity, only outbound as done by `ListAllowedOutboundServicesForIdentity`.

Reply:
This is just removing erroneous code. It previously called ListAllowedServices, then did nothing. Now we call ListMeshServicesForIdentity, which under the hood calls  ListAllowedServices, but combines with the Apex services from traffic splits.

Reply:
Would you mind fixing the erroneous code in a separate PR considering it is outside the scope of multicluster. This previously called `ListAllowedOutboundServicesForIdentity()`, not sure what you mean why you say `then did nothing`. Clearly we should not be programming upstream clusters for all services corresponding to the identity.

Reply:
it doesn't.. take a look at the implementation of ListMeshServicesForIdentity. but ok will send anotehr PR

Reply:
No, I the understand the bug, but your fix doesn't seem not correct either. Should be calling `ListAllowedOutboundServicesForIdentity` instead of `ListMeshServicesForIdentity`